sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -379,22 +374,35 @@ private[hive] class HiveClientImpl(
           s"Hive ${version.fullVersion} does not support altering database location")
       }
     }
-    client.alterDatabase(
+    val hiveDb = toHiveDatabase(database, false)
+    client.alterDatabase(database.name, hiveDb)
+  }
+
+  private def toHiveDatabase(database: CatalogDatabase, isCreate: Boolean): HiveDatabase = {
+    val props = database.properties
+    val dbOwner = props.getOrElse(PROP_OWNER_NAME, if (isCreate) userName else null)
+    val dbOwnerType = props.getOrElse(PROP_OWNER_TYPE, "USER")
```

Comment:
to be consistent, how about `props.getOrElse(PROP_OWNER_TYPE, if (isCreate) "USER" else null)`

Reply:
sgtm

Reply:
let's keep it for enumeration

Reply:
I see the problem now. Can we skip setting owner if it's not in the properties? e.g.
```
props.get(PROP_OWNER_NAME).orElse(if (isCreate) Some(userName) else None).foreach { ownerName =>
  shim.setDatabaseOwnerName(hiveDb, ownerName)
}
```

Reply:
OK